### PR TITLE
fix: improve test coverage for decimal to primitive type casts

### PR DESCRIPTION
**Pull request by parthchandra**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion-comet/issues/286

## Rationale for this change

Validates there is sufficient test coverage for casting decimal type to primitive types.

## What changes are included in this PR?

New tests for decimal to floating point types at maximum precision and decimal to boolean

## How are these changes tested?

New tests

### CODE REVIEW

File: spark/src/test/scala/org/apache/comet/CometCastSuite.scala [hunk 1]
```diff
@@ -637,6 +637,18 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
+    castTest(generateDecimalsPrecision38Scale18(), DataTypes.FloatType)
+  }
+
+  test("cast DecimalType(38,18) to DoubleType") {
+    castTest(generateDecimalsPrecision38Scale18(), DataTypes.DoubleType)
+  }
+
+  test("cast DecimalType(38,18) to BooleanType") {
+    castTest(generateDecimalsPrecision38Scale18(), DataTypes.BooleanType)
+  }
```
Comment (coderfender) [inline spark/src/test/scala/org/apache/comet/CometCastSuite.scala:701]: We might also want to add some edge cases here to gain more confidence

File: spark/src/test/resources/sql-tests/expressions/cast/cast_decimal_to_primitive.sql [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+SELECT cast(cast(1.50 as decimal(10,2)) as float),
+       cast(cast(0.00 as decimal(10,2)) as float),
+       cast(cast(-1.50 as decimal(10,2)) as float),
+       cast(cast(NULL as decimal(10,2)) as float)
+
+-- literal casts: decimal(10,2) to boolean
+query
+SELECT cast(cast(1.50 as decimal(10,2)) as boolean),
+       cast(cast(0.00 as decimal(10,2)) as boolean),
+       cast(cast(NULL as decimal(10,2)) as boolean)
```
Comment (coderfender) [inline spark/src/test/resources/sql-tests/expressions/cast/cast_decimal_to_primitive.sql:169]: Perhaps we could improve variability by adding more scale / precision options

**Review by coderfender — COMMENTED** [review]
Minor comments .Thank you for adding more tests @parthchandra

File: native/spark-expr/src/conversion_funcs/numeric.rs [hunk 3]
```diff
@@ -67,11 +67,15 @@ pub(crate) fn is_df_cast_from_decimal_spark_compatible(to_type: &DataType) -> bo
...
-            | DataType::Float32
-            | DataType::Float64
+            | DataType::Float32 // DataFusion divides i128 by 10^scale in f64, then narrows to
+            | DataType::Float64 // f32 if needed; empirically matches Spark's BigDecimal.doubleValue
+                                // / floatValue for all tested values
             | DataType::Decimal128(_, _)
             | DataType::Decimal256(_, _)
             | DataType::Utf8 // note that there can be formatting differences
+                             // Note: Boolean is intentionally absent. Decimal-to-boolean uses a dedicated
+                             // spark_cast_decimal_to_boolean function (in cast.rs) that checks the raw i128
```
Comment (coderfender) [inline native/spark-expr/src/conversion_funcs/numeric.rs]: Thank you for adding this useful comment

File: spark/src/test/resources/sql-tests/expressions/cast/cast_decimal_to_primitive.sql [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  (0.00, 0),
+  (0.01, 1),
+  (-0.01, -1),
+  (99999999.99, 99999),
+  (-99999999.99, -99999),
+  (NULL, NULL)
+
+-- decimal(10,2) column to FLOAT
+query
+SELECT cast(d10 as float) FROM test_cast_decimal
```
Comment (comphead) [inline spark/src/test/resources/sql-tests/expressions/cast/cast_decimal_to_primitive.sql:36]: should we also cast to int/longs?

**Comment by parthchandra** [comment]
@comphead @coderfender added more test cases.

**Comment by parthchandra** [comment]
investigating the ci failure ...

**Comment by parthchandra** [comment]
@comphead @coderfender - updated to address your comments

**Review by andygrove — APPROVED** [review]
Thanks for adding more tests @parthchandra!

**Comment by parthchandra** [comment]
Merged. Thanks @andygrove @coderfender @comphead !
